### ISSUE TITLE
golangci-lint: iterBinarySearch and binarySearch are unused

### ISSUE THREAD

**Issue by cclauss** [TheAlgorithms/Go#96]
__golangci-lint run --no-config searches__
```
##[error]searches/binary_search.go:20:6: `iterBinarySearch` is unused (deadcode)
func iterBinarySearch(array []int, target int, lowIndex int, highIndex int) int {
     ^
##[error]searches/binary_search.go:6:6: func `binarySearch` is unused (unused)
func binarySearch(array []int, target int, lowIndex int, highIndex int) int {
     ^
```

**Comment by tjgurwara99** [comment]
This is because the names are not capitalised. Simple fix is to just capitalise the function names so that they are accessible in outside/external(? - not sure about the terminology).

**Comment by tjgurwara99** [comment]
This is my personal opinion, but I think the repository can be made much better if we use the standard Golang restrictions on how things are done. For example, from what I understand there are a few things that should be taken into account:

1. The first step, to clean up this whole thing would be to (re)write code so that there is no main function in any file.
2. The main thing that people want to contribute - for example `binarySearch` algorithm - must be capitalised. That way that program will not need a `main` function. Any other functions that are being used inside the `binarySearch` (in this case) can be left without the first letter capitalised, this would not throw the `deadcode`  or `unused` errors.
3. Each function must have a test file with it. So that the go package can run automated tests based on some parsing (to make it automated in the workflow - you will not need `|| True` if this is set up properly). 
4. Finally, from my experience, Go has been developed with modularity and parallelisation in mind. Therefore there are restrictions in the way that people can develop Go code. These restrictions include the idea that a function should not be redeclared anywhere - it actually causes issues. This way, people would have to see whether a function that they need for their new algorithm is already declared and if so can use that function in their algorithm without having to create their own new function. Of course, they can if they want to with function with the first letter not capitalised inside that particular file. These restrictions should be mentioned in a Contribution Guideline.

Anyways, not sure if you'd read this as this is a very old issue but there were my thoughts on how you guys can go about improving this repository. Let me know if I can help with something😄

**Comment by cclauss** [comment]
@tjgurwara99 Can this issue now be closed?

**Comment by tjgurwara99** [comment]
Yeah it's fixed in the PR I submitted

**Comment by cclauss** [comment]
If you add the magic text
Fixes: #96
-- or --
Closes #96
to your comment message then GitHub will autoclose issue #96 when your PR gets merged.

Thanks for the awesome work!
